### PR TITLE
libjpeg: rename libjpeg.dll.lib to libjpeg.lib when setting.compiler=…

### PR DESCRIPTION
**Pull request by ledocc**
Specify library name and version:  **libjpeg/9c**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

…msvc and option.shared=True

libjpeg.dll.lib name come from our patch, and this is not the regular name of  jpeg .lib file.
Cmake’s FindJPEG.cmake script will not found it, so just rename to libjpeg.lib to be
consistent with other jpeg implemention.

### CODE REVIEW

**Comment by uilianries** [comment]
> libjpeg.dll.lib name come from our patch

Is it possible to change the patch instead?

**Comment by ledocc** [comment]
This is my first option, but I choose less intrusive way.
I do that asap.

**Comment by ledocc** [comment]
help

**Comment by uilianries** [comment]
Looking it again, I don't think it's wrong, because libjpeg.dll.lib is the layer for liking libjpeg.dll on runtime. Actually .dll.lib is very regular for msvc + shared builds. If FindJPEG.cmake is not working, we should check what's wrong related to the required library, but the name is normal based on other recipes.

**Comment by Croydon** [comment]
@uilianries You had doubts if this is the right way, but approved 🤔  An accident? 😄

**Comment by uilianries** [comment]
@Croydon Accident, unfortunately, I can't remove a review.

**Comment by SSE4** [comment]
my 5 cents: if we produce binaries which are incompatible with `find_package` or other de-facto standards, that's huge downside. the intention is to provide binaries which everyone could integrate transparently into the existing projects. I'd say, our patch shouldn't change names produced, and `test_package` should then include the code to validate binaries could be discovered.

**Comment by SSE4** [comment]
as for `find_package`, check https://github.com/Kitware/CMake/blob/master/Modules/FindJPEG.cmake#L61
it uses regular [find_library](https://cmake.org/cmake/help/latest/command/find_library.html), which uses [list](https://github.com/Kitware/CMake/blob/master/Modules/FindJPEG.cmake#L55) of names: jpeg jpeg-static libjpeg libjpeg-static.
for suffixes, under the hood it would use [CMAKE_IMPORT_LIBRARY_SUFFIX](https://cmake.org/cmake/help/v3.0/variable/CMAKE_IMPORT_LIBRARY_SUFFIX.html).
which is set to:
- .dll.a for MinGW https://github.com/Kitware/CMake/blob/master/Modules/Platform/Windows-GNU.cmake#L17
- .lib for MSVC https://github.com/Kitware/CMake/blob/07cfb18f9d29cfc0588ede928846a03ec5599c48/Modules/Platform/Windows.cmake#L16

as result, `find_package` would never use **.dll.lib** suffix, nor for MSVC nor for MinGW.
and as libjpeg itself never produces **.dll.lib** (it's only our recipe invention), I suggest we shouldn't do that.

yes, there are cases when **.dll.lib** is produced, e.g. libiconv. but that's library itself intentionally doing that.

**Comment by uilianries** [comment]
I agree with @SSE4

**Comment by ledocc** [comment]
@Croydon SSE4 and uilianries are agree, so can you merge this PR ? or more work is required ?

**Comment by SSE4** [comment]
@ledocc the process is simple - the PR merged as soon as it gets 3 approvals. as there are many activity, it sometimes needed to wait a bit until PR gets an approval (or review).

**Review by uilianries — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**
